src/node/factory.ts

```diff
@@ -64,16 +71,26 @@ export class TurboFactory extends TurboBaseFactory {
             logger: this.logger,
           });
 
+    token ??= 'arweave'; // default to arweave if token is not provided
+    if (!tokenTools) {
+      tokenTools = tokenMap[token]?.({
+        gatewayUrl,
+        logger: this.logger,
+      });
+    }
```

Comment:
if token does not exist by now, being provided by user or derived from signer, then use arweave. sets the token tools if they exist for the token. if it doesn't, all othr methods that dont depend on token tools will still work